File: apiwzsk/WzskQPrsADetail.cpp
/**
	* \file WzskQPrsADetail.cpp
	* API code for table TblWzskQPrsADetail (implementation)
	* \copyright (C) 2016-2020 MPSI Technologies GmbH
	* \author Alexander Wirthmueller (auto-generation)
	* \date created: 1 Jul 2025
  */
// IP header --- ABOVE

#include "WzskQPrsADetail.h"

using namespace std;
using namespace Sbecore;
using namespace Xmlio;

/******************************************************************************
 class WzskQPrsADetail
 ******************************************************************************/

WzskQPrsADetail::WzskQPrsADetail(
			const uint jnum
			, const string x1SrefKType
			, const string titX1SrefKType
			, const string Val
		) :
			jnum(jnum)
			, x1SrefKType(x1SrefKType)
			, titX1SrefKType(titX1SrefKType)
			, Val(Val)
		{
};

bool WzskQPrsADetail::readXML(
			xmlXPathContext* docctx
			, string basexpath
			, bool addbasetag
		) {
	bool basefound;

	if (addbasetag)
		basefound = checkUclcXPaths(docctx, basexpath, basexpath, "WzskQPrsADetail");
	else
		basefound = checkXPath(docctx, basexpath);

	if (basefound) {
		extractStringUclc(docctx, basexpath, "x1SrefKType", "typ", x1SrefKType);
		extractStringUclc(docctx, basexpath, "titX1SrefKType", "typ2", titX1SrefKType);
		extractStringUclc(docctx, basexpath, "Val", "val", Val);
	};

	return basefound;
};

/******************************************************************************
 class ListWzskQPrsADetail
 ******************************************************************************/

ListWzskQPrsADetail::ListWzskQPrsADetail() {
};

ListWzskQPrsADetail::ListWzskQPrsADetail(
			const ListWzskQPrsADetail& src
		) {
	nodes.resize(src.nodes.size(), NULL);
	for (unsigned int i = 0; i < nodes.size(); i++) nodes[i] = new WzskQPrsADetail(*(src.nodes[i]));
};

ListWzskQPrsADetail& ListWzskQPrsADetail::operator=(
			const ListWzskQPrsADetail& src
		) {
	nodes.resize(src.nodes.size(), NULL);
	for (unsigned int i = 0; i < nodes.size(); i++) nodes[i] = new WzskQPrsADetail(*(src.nodes[i]));

	return *this;
};

ListWzskQPrsADetail::~ListWzskQPrsADetail() {
	clear();
};

void ListWzskQPrsADetail::clear() {
	for (unsigned int i = 0; i < nodes.size(); i++) if (nodes[i]) delete nodes[i];
	nodes.resize(0);
};

bool ListWzskQPrsADetail::readXML(
			xmlXPathContext* docctx
			, string basexpath
			, bool addbasetag
		) {
	bool basefound;

	vector<unsigned int> ics;
	vector<bool> shorttags;

	WzskQPrsADetail* rec = NULL;

	string s;

	if (addbasetag)
		basefound = checkUclcXPaths(docctx, basexpath, basexpath, "ListWzskQPrsADetail");
	else
		basefound = checkXPath(docctx, basexpath);

	extractList(docctx, basexpath, "WzskQPrsADetail", "row", "jnum", ics, shorttags);

	clear();

	for (unsigned int i = 0; i < ics.size(); i++) {
		rec = new WzskQPrsADetail(ics[i]);

		if (shorttags[i]) s = basexpath + "/row[@jnum='" + to_string(ics[i]) + "']";
		else s = basexpath + "/WzskQPrsADetail[@jnum='" + to_string(ics[i]) + "']";

		if (rec->readXML(docctx, s)) nodes.push_back(rec);
		else delete rec;
	};

	return basefound;
};

File: sapiwzsk/WzskQPrsADetail.swift
/**
	* \file WzskQPrsADetail.swift
	* Swift API code for table TblWzskQPrsADetail
	* \copyright (C) 2022 MPSI Technologies GmbH
	* \author Alexander Wirthmueller (auto-generation)
	* \date created: 25 Mar 2022
  */
// IP header --- ABOVE

import Foundation
import sbecore

/**
	* WzskQPrsADetail
	*/
public class WzskQPrsADetail {

	public init(
				_ jnum: Int = 0
				, _ x1SrefKType: String = ""
				, _ titX1SrefKType: String = ""
				, _ Val: String = ""
			) {
		self.jnum = jnum
		self.x1SrefKType = x1SrefKType
		self.titX1SrefKType = titX1SrefKType
		self.Val = Val
	}

	public var jnum: Int = 0
	public var x1SrefKType: String
	public var titX1SrefKType: String
	public var Val: String

	public func readXML(
				_ doc: XMLDocument
				, _ _basexpath: String
				, _ addbasetag: Bool
			) -> Bool {
		var valid: Bool
		var dummy: Set<Int> = []
		var basexpath: String? = _basexpath

		if addbasetag {basexpath = Xmlio.checkUclcXPaths(doc, _basexpath, "WzskQPrsADetail")}
		valid = basexpath != nil

		if valid {valid = Xmlio.checkXPath(doc, basexpath!)}

		if valid {
			x1SrefKType = Xmlio.extractStringUclc(doc, basexpath!, "x1SrefKType", "typ", &dummy, 0)
			titX1SrefKType = Xmlio.extractStringUclc(doc, basexpath!, "titX1SrefKType", "typ2", &dummy, 0)
			Val = Xmlio.extractStringUclc(doc, basexpath!, "Val", "val", &dummy, 0)
		}
		
		return valid
	}
}

/**
	* ListWzskQPrsADetail
	*/
public class ListWzskQPrsADetail {
	public func clear() {
		nodes.removeAll()
	}

	public var nodes: [WzskQPrsADetail] = []
	
	public func readXML(
				_ doc: XMLDocument
				, _ _basexpath: String
				, _ addbasetag: Bool
			) -> Bool {
		var basexpath: String? = _basexpath

		var ics: [Int] = []
		var shorttags: [Bool] = []

		var rec: WzskQPrsADetail

		var s: String

		if addbasetag {basexpath = Xmlio.checkUclcXPaths(doc, _basexpath, "ListWzskQPrsADetail")}

		if basexpath != nil {
			if (Xmlio.checkXPath(doc, basexpath!)) {
				Xmlio.extractList(doc, basexpath!, "WzskQPrsADetail", "row", "jnum", &ics, &shorttags)

				clear()
				
				for i in 0 ... ics.count - 1 {
					rec = WzskQPrsADetail()

					if shorttags[i] {s = basexpath! + "/row[@jnum='" + String(ics[i]) + "']"}
					else {s = basexpath! + "/WzskQPrsADetail[@jnum='" + String(ics[i]) + "']"}
					
					if rec.readXML(doc, s, false) {nodes.append(rec)}
				}
			}
			
			return true
		}

		return false
	}
}
